addons/point_of_sale/models/pos_session.py

```diff
@@ -646,7 +646,7 @@ def _create_picking_at_end_of_session(self):
             pickings.write({'pos_session_id': self.id, 'origin': self.name})
 
     def _create_balancing_line(self, data, balancing_account, amount_to_balance):
-        if (not float_is_zero(amount_to_balance, precision_rounding=self.currency_id.rounding)):
+        if not self.company_id.currency_id.is_zero(amount_to_balance):
             balancing_vals = self._prepare_balancing_line_vals(amount_to_balance, self.move_id, balancing_account)
```

Comment:
```py
    def _prepare_balancing_line_vals(self, imbalance_amount, move, balancing_account):
        partial_vals = {
            'name': _('Difference at closing PoS session'),
            'account_id': balancing_account.id,
            'move_id': move.id,
            'partner_id': False,
        }
        # `imbalance_amount` is already in terms of company currency so it is the amount_converted
        # param when calling `_credit_amounts`. amount param will be the converted value of
        # `imbalance_amount` from company currency to the session currency.
        imbalance_amount_session = 0
        if (not self.is_in_company_currency):
            imbalance_amount_session = self.company_id.currency_id._convert(imbalance_amount, self.currency_id, self.company_id, fields.Date.context_today(self))
        return self._credit_amounts(partial_vals, imbalance_amount_session, imbalance_amount)
```
The `balancing_account` must be expressed in foreign currency so it seems the previous code was the correct one.

I guess `_create_balancing_line` is the method used to balance the closing entry. The only way to solve the problem is to really sum all balance (not just the balance in foreign currency) and not apply the currency rate for this line. This is what we do on an invoice to balance the journal entry.

Reply:
@caburj It isn't. Look at this line:
`imbalance_amount_session = self.company_id.currency_id._convert(imbalance_amount, self.currency_id, self.company_id, fields.Date.context_today(self))`
`imbalance_amount` is clearly expressed in foreign currency here and this method is there to get the amount in company currency. However, for the payment terms, you want to close both amounts, not to apply the rate. It can't work correctly that way.

Reply:
@smetl The `imbalance_amount` is in the company currency, and the `imbalance_amount_session` is in the session currency. I couldn't understand what do you mean by "foreign currency".

Reply:
@pedrambiria My bad.